## <span style="font-family:serif"> Abstract </span>

> <span style="font-family:serif"> Crunchyroll is a streaming platform that offers anime, manga, and Asian dramas to users worldwide. A dataset was found on Kaggle, a platform for users to access and download public datasets, that contained information pertaining to anime found in the streaming service including user ratings and genre information. Because Cruncyroll has over 1000 anime titles to offer to a user base of more than 100 million registered users, the data generated by the streaming service is immense and provides valuable insights into the preferences and behaviors of anime fans.
</span>

> <span style="font-family:serif"> The main aim of this project is to analyze the data using visualization tools to identify any possible patterns that can be found in Crunchyroll from 2020. The analysis would involve examining user ratings, the weight of the votes, and genre information for different anime. The results of this analysis can be used by Crunchyroll to improve user experience, personalize recommendations, and inform content creation decisions. </span>

---

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.linear_model import LinearRegression
%matplotlib inline

### read_csv is a pandas method that allows us to read a comma-seperated value (csv) file
anime_data = pd.read_csv(r'anime.csv')

In [2]:
genres = [] ### initialize an emtpy array
substring = "genre" ### create a substring "genre"

for col in anime_data.columns: ### for each column in the dataframe
    if substring in col:       ### check if "genre" is in the name of the column
        genres.append(col)     ### add that column name to the array

In [3]:
# initialize an empty list to store the calculated average value of the rates
avg_rating = []

# for each genre listed
for genre in genres: 
    # calculate the averate rating for anime that are classified as that genre
    avg_rating.append(anime_data.loc[anime_data[genre] == 1, 'rate'].mean()) 
    
# find the value of the 5 genres with the highest rating
top_5 = sorted(avg_rating, reverse=True)[:5]

index = []
for avg in top_5:
    index.append(avg_rating.index(avg))
    
index

[12, 6, 9, 26, 21]

In [4]:
# for each top genre
for idx in index:
    # list the average rating of anime classified as that genre
    print("mean of "+ genres[idx] + ": " + str(anime_data.loc[anime_data[genres[idx]] == 1, 'rate'].mean()))

mean of genre_jdrama: 4.77
mean of genre_food: 4.7
mean of genre_horror: 4.63
mean of genre_sports: 4.39448275862069
mean of genre_seinen: 4.281313131313133


In [5]:
# for each top genre
for idx in index:
    # list how many anime are classified as that genre
    print("# of "+ genres[idx] + ": " + str(anime_data.loc[anime_data[genres[idx]] == 1, genres[idx]].sum()))

# of genre_jdrama: 3.0
# of genre_food: 1.0
# of genre_horror: 1.0
# of genre_sports: 58.0
# of genre_seinen: 99.0


In [32]:
genre_count = []
genre_mean = []
# for each genre
for genre in genres:
    # list how many anime are classified as that genre
    genre_count.append(anime_data.loc[anime_data[genre] == 1, genre].sum())
    
    # list how many anime are classified as that genre
    genre_mean.append(anime_data.loc[anime_data[genre] == 1, 'rate'].mean())


genre_info = pd.DataFrame([genres, genre_mean, genre_count]).transpose()
genre_info.rename(columns = { 0:'genre', 1:'mean', 2:'count' })

[417.0, 222.0, 464.0, 329.0, 2.0, 301.0, 1.0, 88.0, 56.0, 1.0, 34.0, 37.0, 3.0, 22.0, 5.0, 70.0, 52.0, 69.0, 28.0, 276.0, 220.0, 99.0, 4.0, 40.0, 120.0, 325.0, 58.0, 197.0, 63.0]
[3.917194244604316, 3.9193693693693685, 4.205064655172415, 4.220000000000001, 3.065, 3.976777408637873, 4.7, 4.045454545454548, 4.0878571428571435, 4.63, 3.8911764705882357, 3.9527027027027013, 4.77, 4.007727272727274, 2.686, 3.9344285714285716, 4.058076923076924, 4.098985507246377, 4.0525, 4.149094202898552, 3.8333181818181825, 4.281313131313133, 3.47, 4.04375, 4.104000000000002, 4.253969230769228, 4.39448275862069, 4.1279187817258896, 4.07047619047619]


genre     mean count
0             genre_action  3.91719   417
1          genre_adventure  3.91937   222
2             genre_comedy  4.20506   464
3              genre_drama     4.22   329
4             genre_family    3.065     2
5            genre_fantasy  3.97678   301
6               genre_food      4.7     1
7              genre_harem  4.04545    88
8         genre_historical  4.08786    56
9             genre_horror     4.63     1
10             genre_idols  3.89118    34
11            genre_isekai   3.9527    37
12            genre_jdrama     4.77     3
13     genre_magical_girls  4.00773    22
14      genre_martial_arts    2.686     5
15             genre_mecha  3.93443    70
16             genre_music  4.05808    52
17           genre_mystery  4.09899    69
18  genre_post_apocalyptic   4.0525    28
19           genre_romance  4.14909   276
20            genre_sci_fi  3.83332   220
21            genre_seinen  4.28131    99
22           genre_sgdrama     3.47     4
23             genre_shojo  4.04375    40
24            genre_shonen    4.104   120
25     genre_slice of life  4.25397   325
26            genre_sports  4.39448    58
27      genre_supernatural  4.12792   197
28          genre_thriller  4.07048    63

# Notes for Analysis

- the top 5 anime genre from highest average rating to lowest average rating Jdrama, Food, Horror, Sports, and Seinen.
- jdrama avg = 4.77
- food avg = 4.7
- horror avg = 4.63
- sports avg = 4.39
- sports avg = 4.28

# Notes After Analysis



- One more thing to note is that the data from this dataset was crawled and posted on Kaggle in 2020. This Crunchyroll data is very likely to not be up-to-date and may or may not representative of current Crunchyroll information if more reviews and ratings were added to various anime in the dataset.

- Year the anime started.
- Manga vs Animated Show.
- Rating.
- Dataset was from 2020, may be outdated compared to data currently.

- Take note of anime I am interested in, perhaps what studios that animated those series, and/or the year they were made.

- It's a little difficult to determine what is what from this database. Some could be anime series, others could be manga, some could be live actions, and some might be shows that aren't animated at all and are vaguely related to anime like podcasts, or projects that take inspiration from anime. A next step for this project would likely be to better organize the data and/or to curate the dataset to contain more comprehensive information.






In [7]:
####
#### The data is incomplete/inaccurate some anime do not have listed genre, so we must correct the data manually to be more complete
####

no_genre = [] ### initialize an empty list for anime that do not have any genre information

for anime in anime_data['anime']: # for each anime in the data frame
    
    genre_count = 0 ### a temporary variable
    for genre in genres: ### check each genre in the "genres" array
        
        ### check each genre column and check if the anime is of said genre
        if anime_data.loc[anime_data['anime'] == anime][genre].sum() == True:      
            genre_count += 1 
            break
    
    ### if there is no genre information append to "no_genre" list
    if genre_count == 0:
        no_genre.append(anime)
            
no_genre    

['Naruto Shippuuden',
 'Shugo Chara',
 'BLEACH',
 'Naruto',
 'REBORN!',
 'Gintama',
 'Hayate the Combat Butler! (S1 e S2)',
 'Blue Exorcist',
 'Durarara!!',
 'Time of Eve',
 'Linebarrels of Iron',
 'Hunter x Hunter',
 'Inu X Boku Secret Service',
 'A Dark Rabbit has Seven Lives',
 'Fate/Zero',
 "JoJo's Bizarre Adventure",
 'Chihayafuru',
 'RWBY',
 'Gankutsuou',
 'Fairy Tail',
 'Level E',
 'Rio Rainbow Gate',
 'Horizon in the Middle of Nowhere',
 'Kids on the Slope',
 'NARUTO Spin-Off: Rock Lee & His Ninja Pals',
 'Galaxy Express 999',
 "Leiji Matsumoto's OZMA",
 'Do You Love Your Mom and Her Two-Hit Multi-Target Attacks?',
 'Os Cavaleiros do Zodíaco - Ômega',
 'Phi Brain',
 'SHIROBAKO',
 'Digimon Xros Wars - The Young Hunters Who Leapt Through Time',
 'La storia della Arcana Famiglia',
 'Death Note (Drama)',
 'Mischievous Kiss - Love in Tokyo',
 'Kaiji',
 'Sakura Quest',
 'makuranodanshi',
 "The Ryuo's Work is Never Done!",
 'Tokyo Ghoul: re',
 'Anifile',
 'gdgd Fairies',
 'Cuticle Det

This block of code goes through the data, to check whether or not there is genre information for all of the shows listed. To clean the data, I would remove the shows that have no information on genre because it would be hard to conduct analyses. However, from my prior knowledge of anime, I was shocked to find that popoular shows like "Naruto", "BLEACH", and Shugo Chara did not have any genre information. Rather than simply removing possibly impactful data points, I am researching various resources to try fix these gaps in data. So far MyAnimeList.net has played a large role in restoring the data.

By running this code, I am providing myself with a reference list of the anime data that needs to be cleaned. Part of this process requires me to research the anime and its genre and manually enter the appropriate information into the csv file. Although it is true that I would need to either fill the missing genre information or remove it, it is also very likely that I would need to go through all of the data to verify that the genre information is accurate.

In [8]:
anime_data

anime  \
0                        Naruto Shippuuden   
1                              Shugo Chara   
2                                   BLEACH   
3                                   Naruto   
4                               Skip Beat!   
...                                    ...   
1250                 Space Warrior Baldios   
1251  Captain Harlock: Arcadia of my Youth   
1252                          Osomatsu-kun   
1253                 Magic User's Club OVA   
1254         Izumo: Flash of a Brave Sword   

                                              anime_url  \
0          https://www.crunchyroll.com/naruto-shippuden   
1               https://www.crunchyroll.com/shugo-chara   
2                    https://www.crunchyroll.com/bleach   
3                    https://www.crunchyroll.com/naruto   
4                 https://www.crunchyroll.com/skip-beat   
...                                                 ...   
1250  https://www.crunchyroll.com/space-warrior-baldios   
1251  https://www.crunchyroll.com/captain-harlock-ar...   
1252           https://www.crunchyroll.com/osomatsu-kun   
1253   https://www.crunchyroll.com/magic-users-club-ova   
1254  https://www.crunchyroll.com/izumo-flash-of-a-b...   

                                              anime_img  episodes  votes  \
0     https://img1.ak.crunchyroll.com/i/spire4/1c1df...       500  48518   
1     https://img1.ak.crunchyroll.com/i/spire1/7c0c0...        51  33688   
2     https://img1.ak.crunchyroll.com/i/spire2/52edb...       366  23900   
3     https://img1.ak.crunchyroll.com/i/spire2/01453...       220  16613   
4     https://img1.ak.crunchyroll.com/i/spire3/4bbe3...         0  15011   
...                                                 ...       ...    ...   
1250  https://img1.ak.crunchyroll.com/i/spire4/e6897...         0      0   
1251  https://img1.ak.crunchyroll.com/i/spire1/fc4f9...         0      0   
1252  https://img1.ak.crunchyroll.com/i/spire4/b0778...         0      0   
1253  https://img1.ak.crunchyroll.com/i/spire4/3371b...         0      0   
1254  https://img1.ak.crunchyroll.com/i/spire1/94ec4...         0      0   

      weight  rate  rate_1  rate_2  rate_3  ...  genre_romance  genre_sci_fi  \
0     233346  4.81    1111     275     817  ...            0.0           0.0   
1     163641  4.86     503     104     438  ...            0.0           0.0   
2     115445  4.83     391      98     370  ...            0.0           0.0   
3      79667  4.80     456      64     282  ...            0.0           0.0   
4      73683  4.91      98      20     112  ...            1.0           0.0   
...      ...   ...     ...     ...     ...  ...            ...           ...   
1250       0  0.00       0       0       0  ...            0.0           1.0   
1251       0  0.00       0       0       0  ...            0.0           1.0   
1252       0  0.00       0       0       0  ...            0.0           0.0   
1253       0  0.00       0       0       0  ...            1.0           1.0   
1254       0  0.00       0       0       0  ...            0.0           0.0   

      genre_seinen  genre_sgdrama  genre_shojo  genre_shonen  \
0              0.0            0.0          0.0           0.0   
1              0.0            0.0          0.0           0.0   
2              0.0            0.0          0.0           0.0   
3              0.0            0.0          0.0           0.0   
4              0.0            0.0          0.0           0.0   
...            ...            ...          ...           ...   
1250           0.0            0.0          0.0           0.0   
1251           0.0            0.0          0.0           0.0   
1252           0.0            0.0          0.0           1.0   
1253           0.0            0.0          0.0           0.0   
1254           0.0            0.0          0.0           0.0   

      genre_slice of life  genre_sports  genre_supernatural  genre_thriller  
0                     0.0           0.0               